runtime/Makefile

```diff
@@ -16,6 +16,18 @@ ENABLE_OPENQASM?=OFF
 LIGHTNING_GIT_TAG_VALUE?=latest_release
 LIGHTNING_KOKKOS_GIT_TAG_VALUE?=latest_release
 
+# Assumption: uname is installed in the machine
+# building Catalyst. This assumption will not hold
+# on windows machines, for that we will need to have
+# an extra condition before trying to run uname.
+UNAME := $(shell uname)
+ifeq ($(UNAME),Linux)
+  NPROC = $(shell nproc)
+else ifeq ($(UNAME),Darwin)
+  NPROC = $(shell sysctl -n hw.physicalcpu)
+else
+  NPROC = 1
+endif
```

Comment:
Could this block be replaced by a Python call to https://docs.python.org/3/library/os.html#os.sched_getaffinity  or `python -c "import os; print(os.cpu_count())"` which abstracts away the OS specifics?

Reply:
Sounds good!